### PR TITLE
Added occurrence count

### PR DESCRIPTION
**Pull request by cpursley**
Functional is one way to skin the cat

### CODE REVIEW

**Comment by lolmaus** [comment]
1. Please remove parentheses where they aren't necessary.
2. Please add the standard warning on extending native prototypes.

**Comment by michaelglass** [comment]
3.  and warning that this function doesn't escape regexp characters (e.g. `"^^^".occurences("^") ==  1`
`"[".occurnces("[")` raises an exception

File: chapters/strings/finding-substrings.md [hunk 1]
```diff
@@ -24,8 +24,14 @@ message.indexOf "This", 5
 message.lastIndexOf "This"
 # => 49
 
+# Count occurrences of a given string within a message
+String::occurrences = (search) ->
+  regex = new RegExp(search, 'g')
+  (@match(regex) or []).length
```
Comment (sukima) [inline chapters/strings/finding-substrings.md]: This is not very performant. Your needlessly creating an empty array. How about this:

```coffee
regex.test(this)
```

Comment (sukima) [inline chapters/strings/finding-substrings.md]: How about this:

```coffee
@match(regex)?.length or 0
```

File: chapters/strings/finding-substrings.md [hunk 2]
```diff
@@ -24,8 +24,14 @@ message.indexOf "This", 5
 message.lastIndexOf "This"
 # => 49
 
+# Count occurrences of a given string within a message
+String::occurrences = (search) ->
+  regex = new RegExp(search, 'g')
```
Comment (sukima) [inline chapters/strings/finding-substrings.md]: To escape regexp characters try this (courtesy of [this SO answer](http://stackoverflow.com/a/3561711/227176)):

```coffee
search = search.replace(/[-\/\\^$*+?.()|[\]{}]/g, "\\$&")
```

**Comment by cpursley** [comment]
Thanks for the suggestions; I'll think through them and propose an improvement.

**Pull request closed without merging**